### DISCUSSION TITLE
General Discussion

### DISCUSSION THREAD

**Discussion by NTDLS** [NTDLS/Katzebase#84] [Announcements]
## Large code merger of all Katzebase projects and extraction of parser.

@ingted Again, I hate to make big changes but that's kind of where this project is in the development cycle.

I have merged all the Katzebase projects into one solution, this alleviates a lot of contention on updating nuget packages as well as allows us to share libraries across projects.

With this change, the client version has been moved back to the 0.22.0 server version and I intend to keep them in sync.

I have also moved the parser from the engine and into an independent library, I did this so that I can use the parser in the UI for static analysis and code-completion. This also means that you can add tests to the parser all by itself now independent of the engine.

**Comment by ingted** [comment]
Great software engineering!!

**Comment by NTDLS** [comment]
Finally have a reasonably working static analyzer. It still needs some work, but having the query parser in a separate library is VERY nice!
![image](https://github.com/user-attachments/assets/5f0a92a1-b4bf-47b6-8a9b-117af4a640b7)
![image](https://github.com/user-attachments/assets/7b25174f-e685-411e-b182-8de3e6a9ad5c)
![image](https://github.com/user-attachments/assets/4af3bc93-695b-405e-b169-375a0dd81135)
![image](https://github.com/user-attachments/assets/4e17be22-9bf6-495b-8d8b-042ac5c9f8f3)
